specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,503 @@
+
+In addition to these new properties, there are several existing properties that indicate valid values for an input. Currently, these existing properties are enforced on some platforms whose UI controls allow them to be enforced natively in the UI. For example `Input.Date` has `min` and `max` properties. Ideally, this is implemented as a calendar picker in which dates outside the supported range are greyed out. Currently, however, only some platforms support this behavior due to the presence or absence of a native control with that functionality.
+
+With the advent of this input validation feature, those platforms which do not support these properties should use the input validation behavior described in this spec to add support. Enforcing them natively in the UI by preventing the user from picking an invalid value is still preferred, but when that is not feasible, input validation should be used in these cases. Considering this, the full list of properties that should be either enforced or validated are:
+
+| Input | Properties | |
+| --- | --- | --- |
+| `Input.Text` | `regex` <br> `maxLength` | New for this feature <br> ExistingProperty |
+| `Input.Number` | `min`, `max` | Existing Properties |
+| `Input.Date` | `min`, `max` | Existing Properties |
+| `Input.Time` | `min`, `max` | Existing Properties |
+|**All inputs** | `isRequired` | New for this feature |
+
+### Validation success condition
+
+Once we have defined the properties to be validated, we also have to provide some consistency on how we consider every property to be successfully validated (no error is raised):
+
+| Input | Property | Success condition |
+| --- | --- | --- |
+| `Input.Text` | `regex` | Input complies with provided regex. |
+| `Input.Text` | `maxLength` | Input length is less or equal to maxLength. | 
+| `Input.Text` | `isRequired` | Input has any character in it including spaces or any other character. | 
+| `Input.Number` | `min` | Input value is greater or equal to min. |
+| `Input.Number` | `max` | Input value is less or equal to max. |
+| `Input.Number` | `isRequired` | Input has any positive or negative number in it. |
+| `Input.Date` | `min` | Input value is greater or equal to min. |
+| `Input.Date` | `max` | Input value is less or equal to max. |
+| `Input.Date` | `isRequired` | A date has been selected and populates the input. |
+| `Input.Time` | `min` | Input value is greater or equal to min. |
+| `Input.Time` | `max` | Input value is less or equal to max. |
+| `Input.Time` | `isRequired` | A time has been selected and populates the input. |
+| `Input.ChoiceSet` | `isRequired` | Any value has been selected. In the case of multiselect, 1 or more values have been selected. | 
+| `Input.Toggle` | `isRequired` | ValueOn has been selected. |
+
+### Which Inputs to Validate
```

Comment:
>Which Inputs to Validate [](start = 4, length = 24)

I think this section is a little unwieldy now, and honestly it probably was already when i handed it off.  Consider reorganizing this to put what we're actually shipping up front, and maybe move some of the detailed discussion of other options to the appendix and reference them from the text. Ideally someone would be able to come to this document in the future and easily see what the expected behavior is. We don't want to lose the discussion of tradeoffs, but now that we've decided on something we should probably lead with that. 

Also, because we both added to this at different times and with different opinions, i feel like it's maybe not super consistent about it's message on the tradeoffs of the various approaches (a section i wrote says one way is the best, and a section you wrote says something else). Consider re-reading through the whole section and seeing if there are places where we should update the language to present a consistent point of view. Again, we don't want to lose the substance of the discussions/disagreements, but someone reading should be able to come away from it with a clear feeling that "The team decided X approach was best for Y reasons."